ldap/servers/slapd/back-ldbm/db-mdb/mdb_import.c

```diff
@@ -1150,14 +1150,22 @@ process_db2index_attrs(Slapi_PBlock *pb, ImportCtx_t *ctx)
      * TBD
      */
     char **attrs = NULL;
+    char *attrname = NULL;
+    char *pt = NULL;
     int i;
 
     slapi_pblock_get(pb, SLAPI_DB2INDEX_ATTRS, &attrs);
     /* Should perhaps add some tests to check that indexes are valid */
     for (i = 0; attrs && attrs[i]; i++) {
         switch (attrs[i][0]) {
         case 't': /* attribute type to index */
-            slapi_ch_array_add(&ctx->indexAttrs, slapi_ch_strdup(attrs[i] + 1));
+            attrname = slapi_ch_strdup(attrs[i] + 1);
```

Comment:
I wonder if static analysers will scream here for the case where `attrname` might be NULL after `slapi_ch_strdup`. We have the null check in `slapi_ch_strdup`, so probably not... (if they are smart enough)

Reply:
slapi-plugin.h:char *slapi_ch_strdup(const char *s) __ATTRIBUTE__((returns_nonnull));
So it is explicitly said in the header.

Most static analyser tools handles the attributes and anyway those that does not spawn so much false positive that they are mostly useless
Coverity scan is smart enough, by I noticed that sometime adding useless test for NULL returns tends to increase the number of report of possible NULL dereference because it also considers that if return is tested for NULL, that means that the function may returns NULL. (in short you should either test everywhere or nowhere ...)